src/views/BannerSubmission/index.js

```diff
@@ -12,47 +48,484 @@ const style = {
-                <a href="mailto:360@gordon.edu?Subject=Banner Image Submission">
-                  <Button variant="contained" style={style.uploadButton}>
-                    Email the 360 Team
-                  </Button>
-                </a>
-              </Grid>
+  const authenticated = useAuth();
+  const [currentUsername, setCurrentUsername] = useState('');
+  const isOnline = useNetworkStatus();
+  const [isAdmin, setIsAdmin] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  //Solely for banner management
+  const [banners, setBanners] = useState([]);
+  const [newBannerTitle, setNewBannerTitle] = useState('');
+  const [newBannerWebLink, setNewBannerWebLink] = useState('');
+  const [newBannerSortOrder, setNewBannerSortOrder] = useState(-1);
+  const [openBannerActivity, setOpenBannerActivity] = useState(false);
+
+  //Solely for photo functions
+  const [cropperImageData, setCropperImageData] = useState(null); //null if no picture chosen, else contains picture
+  const [photoDialogErrorTimeout, setPhotoDialogErrorTimeout] = useState(null);
+  const [photoDialogError, setPhotoDialogError] = useState(null);
+  const [snackbar, setSnackbar] = useState({ open: false, text: '', severity: '' });
+  const cropperRef = useRef();
+  const [aspectRatio, setAspectRatio] = useState(null);
+
+  const loadBanners = useCallback(async () => {
+    setLoading(true);
+    if (authenticated) {
+      const existingBanners = await cmsService.getSlides();
+      setLoading(false);
+      setBanners(existingBanners);
+    } else {
+    }
+  }, [authenticated]);
+
+  useEffect(() => {
+    loadBanners();
+  }, [authenticated, loadBanners]);
+
+  useEffect(() => {
+    const loadUsername = async () => {
+      const user = await userService.getProfileInfo();
+      setCurrentUsername(user.AD_Username);
+    };
+
+    loadUsername();
+  }, []);
+
+  useEffect(() => {
+    if (authenticated) {
+      setIsAdmin(userService.getLocalInfo().college_role === 'god');
+    }
+  }, [authenticated]);
+
+  function handlePostClick() {
+    setOpenBannerActivity(true);
+  }
+
+  function handleWindowClose() {
+    setOpenBannerActivity(false);
+    setNewBannerTitle('');
+    setNewBannerWebLink('');
+    setCropperImageData(null);
+  }
+
+  const createSnackbar = (text, severity) => {
+    setSnackbar({ open: true, text, severity });
+  };
+
+  /**********************************************************
+  /*Following functions are solely related to photo submission*
+  /**********************************************************/
+
+  async function clearPhotoDialogErrorTimeout() {
+    clearTimeout(photoDialogErrorTimeout);
+    setPhotoDialogErrorTimeout(null);
+    setPhotoDialogError(null);
+  }
+
+  /**
+   * Creates the Photo Dialog message that will be displayed to the user
+   *
+   * @returns {string} The message of the Photo Dialog
+   */
+  function createPhotoDialogBoxMessage() {
+    let message = '';
```

Comment:
Or alternatively, rather than having a variable that gets set based on the if-else branches below and then returned at the end, you could simply return from each if/else block directly.